ui/src/app/pages/profile/profile-page.tsx

```diff
@@ -353,6 +389,18 @@ export const ProfilePage = withUserProfile()(class extends React.Component<
               {usdElement(profile.freeTierDollarQuota - profile.freeTierUsage)}
             </FlexColumn>
           </FlexRow>}
+          <div>
+            <div style={styles.title}>Optional Demographics Survey</div>
+            <Button
+                type={'link'}
+                onClick={async() => {
+                  if (!profileEdits.demographicSurvey) {
+                    await this.setState(fp.set(['profileEdits', 'demographicSurvey'], {}));
```

Comment:
setState is definitely async and it definitely returns a promise per the documentation you linked. I've been bitten by this before.

Reply:
Can you provide a specific pointer?

setState does have what they call an "async form" where the caller provides a callback as the second argument (which is probably a source of much confusion in our codebase), but I'm pretty sure the setState method itself does not return a promise.

Partially verified by ctrl-clicking into a setState call in IntelliJ and seeing the void return type:

<img width="672" alt="Screen Shot 2020-03-18 at 11 46 23 AM" src="https://user-images.githubusercontent.com/51842/76979344-2d60d200-690e-11ea-92c0-c22c16c7d3c9.png">

Reply:
```You may optionally pass an object as the first argument to setState() instead of a function:

setState(stateChange[, callback])

This performs a shallow merge of stateChange into the new state, e.g., to adjust a shopping cart item quantity:

this.setState({quantity: 2})

This form of setState() is also asynchronous, and multiple calls during the same cycle may be batched together.```

This is how we do all our `setState`s.

Reply:
Yep – I think when they say "asynchronous" it's referring to a different 'asynchronous' than https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function.

This post seems to get more to my point: https://stackoverflow.com/a/53409805